app/src/main/java/ch/sdp/vibester/activity/BuzzerScreenActivity.kt

```diff
@@ -43,28 +49,137 @@ class BuzzerScreenActivity : AppCompatActivity() {
         supportActionBar?.hide()
         setContentView(R.layout.activity_buzzer_screen)
 
+        val ctx: Context = this
+
         val getIntent = intent.extras
-        val nPlayers = getIntent?.getInt("Number of players")
+        if (getIntent != null) {
+
+            val nPlayers = getIntent.getInt("Number of players")
 
-        val answer = findViewById<LinearLayout>(R.id.answer)
-        val answerText = findViewById<TextView>(R.id.answerText)
-        answerText.text = "The song was Demo by The Placeholders"
+            val answer = findViewById<LinearLayout>(R.id.answer)
+            answer.visibility=View.INVISIBLE
+            //val answerText = findViewById<TextView>(R.id.answerText)
+            //answerText.text = "The song was Demo by The Placeholders"
 
-        val allPoints = if (nPlayers!=null) {Array<Int>(nPlayers, { i -> 0 }) } else Array<Int>(MAX_N_PLAYERS, {i ->0})
-        val playersFull = getIntent?.getStringArray("Player Names")
-        val players = nPlayers?.let { playersFull?.copyOfRange(0, it) }
+            val allPoints = Array(nPlayers, { i -> 0 })
+            val playersFull = getIntent.getStringArray("Player Names")
+            val players = nPlayers.let { playersFull?.copyOfRange(0, it) }
 
-        val updater = BuzzerScoreUpdater(buzIds, allPoints)
+            val updater = BuzzerScoreUpdater(buzIds, allPoints)
 
-        if (players != null) {
-            buildScores(players, allPoints)
-            buildBuzzers(players, answer)
+            if (players != null) {

```

Comment:
In general, you always have default values for players, so mayber ther is no need to check for that